File: app/src/main/java/id/indosw/xlstojson/xlstojsonstring/ConvertExcelToJson.java
package id.indosw.xlstojson.xlstojsonstring;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

import org.apache.poi.ss.usermodel.Cell;
import org.apache.poi.ss.usermodel.Row;
import org.apache.poi.ss.usermodel.Sheet;
import org.apache.poi.ss.usermodel.Workbook;
import org.apache.poi.xssf.usermodel.XSSFWorkbook;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;

public class ConvertExcelToJson {

    public static List<Customer> readExcelFile(String filePath){
        try {
            FileInputStream excelFile = new FileInputStream(new File(filePath));
            Workbook workbook = new XSSFWorkbook(excelFile);

            Sheet sheet = workbook.getSheet("Customers");
            Iterator<Row> rows = sheet.iterator();

            List<Customer> lstCustomers = new ArrayList<Customer>();

            int rowNumber = 0;
            while (rows.hasNext()) {
                Row currentRow = rows.next();

                // skip header
                if(rowNumber == 0) {
                    rowNumber++;
                    continue;
                }

                Iterator<Cell> cellsInRow = currentRow.iterator();

                Customer cust = new Customer();

                int cellIndex = 0;
                while (cellsInRow.hasNext()) {
                    Cell currentCell = cellsInRow.next();

                    if(cellIndex==0) { // ID
                        cust.setId(String.valueOf(currentCell.getNumericCellValue()));
                    } else if(cellIndex==1) { // Name
                        cust.setName(currentCell.getStringCellValue());
                    } else if(cellIndex==2) { // Address
                        cust.setAddress(currentCell.getStringCellValue());
                    } else if(cellIndex==3) { // Age
                        cust.setAge((int) currentCell.getNumericCellValue());
                    }

                    cellIndex++;
                }

                lstCustomers.add(cust);
            }

            // Close WorkBook
            workbook.close();

            return lstCustomers;
        } catch (IOException e) {
            throw new RuntimeException("FAIL! -> message = " + e.getMessage());
        }
    }

    public static String convertObjects2JsonString(List<Customer> customers) {
        ObjectMapper mapper = new ObjectMapper();
        String jsonString = "";

        try {
            jsonString = mapper.writeValueAsString(customers);
        } catch (JsonProcessingException e) {
            e.printStackTrace();
        }

        return jsonString;
    }

}

File: app/src/main/java/id/indosw/xlstojson/kotlin/xlstojsonstring/ConvertExcelToJson.kt
package id.indosw.xlstojson.kotlin.xlstojsonstring

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import org.apache.poi.ss.usermodel.Cell
import org.apache.poi.ss.usermodel.Row
import org.apache.poi.ss.usermodel.Workbook
import org.apache.poi.xssf.usermodel.XSSFWorkbook
import java.io.File
import java.io.FileInputStream
import java.io.IOException
import java.util.*

object ConvertExcelToJson {
    fun readExcelFile(filePath: String?): List<Customer> {
        return try {
            val excelFile = FileInputStream(File(filePath!!))
            val workbook: Workbook = XSSFWorkbook(excelFile)
            val sheet = workbook.getSheet("Customers")
            val rows: Iterator<Row> = sheet.iterator()
            val lstCustomers: MutableList<Customer> = ArrayList()
            var rowNumber = 0
            while (rows.hasNext()) {
                val currentRow = rows.next()

                // skip header
                if (rowNumber == 0) {
                    rowNumber++
                    continue
                }
                val cellsInRow: Iterator<Cell> = currentRow.iterator()
                val cust = Customer()
                var cellIndex = 0
                while (cellsInRow.hasNext()) {
                    val currentCell = cellsInRow.next()
                    when (cellIndex) {
                        0 -> { // ID
                            cust.id = currentCell.numericCellValue.toString()
                        }
                        1 -> { // Name
                            cust.name = currentCell.stringCellValue
                        }
                        2 -> { // Address
                            cust.address = currentCell.stringCellValue
                        }
                        3 -> { // Age
                            cust.age = currentCell.numericCellValue.toInt()
                        }
                    }
                    cellIndex++
                }
                lstCustomers.add(cust)
            }

            // Close WorkBook
            workbook.close()
            lstCustomers
        } catch (e: IOException) {
            throw RuntimeException("FAIL! -> message = " + e.message)
        }
    }

    fun convertObjects2JsonString(customers: List<Customer?>?): String {
        val mapper = ObjectMapper()
        var jsonString = ""
        try {
            jsonString = mapper.writeValueAsString(customers)
        } catch (e: JsonProcessingException) {
            e.printStackTrace()
        }
        return jsonString
    }
}